backend/src/main/kotlin/org/loculus/backend/controller/FilesController.kt

```diff
@@ -46,11 +46,16 @@ class FilesController(
         @PathVariable fileName: String,
     ): ResponseEntity<Void> {
         val accessionVersion = AccessionVersion(accession, version)
-        val fileId = submissionDatabaseService.getFileId(accessionVersion, fileCategory, fileName)
+        val fileIdAndReleasedAt = submissionDatabaseService.getFileIdAndReleasedAt(
+            accessionVersion,
+            fileCategory,
+            fileName,
+        )
+        val fileId = fileIdAndReleasedAt?.fileId
+        val isPublic = fileIdAndReleasedAt?.releasedAt != null
         if (fileId == null) {
             throw NotFoundException("File not found")
         }
-        val isPublic = filesDatabaseService.isFilePublic(fileId)!!
         if (!isPublic) {
```

Comment:
I see that we are creating a presigned URL to read the file using the `createUrlToReadPrivateFile` - so this means that we never use the getPublicUrl except for in the `buildFileUrls` function? Should we also not use it there as we might now not set the file to public on release correctly?

Reply:
Ah I see you already mentioned this in your comment above - yes I do think we should (temporarily) remove the S3 option for now as it does not fully work and it not being used

Reply:
Or at least we should document that there is a potential issue when using `s3` as we haven't been able to resolve the race-condition